src/object/obj_class.c

```diff
@@ -370,9 +370,33 @@ daos_oclass_id2name(daos_oclass_id_t oc_id, char *str)
 	}
 
 	D_ASSERT(oc->oc_id == OC_UNKNOWN);
+	strcpy(str, "UNKNOWN");
 	return -1;
 }
 
+/** Return the list of registered oclass names */
+size_t
+daos_oclass_names_list(size_t size, char *str)
+{
+	struct daos_obj_class   *oc;
+	size_t len = 0;
+
+	if (size <= 0 || str == NULL)
+		return -1;
+
+	*str = '\0';
+	for (oc = &daos_obj_classes[0]; oc->oc_id != OC_UNKNOWN; oc++) {
+		if (oc->oc_name == NULL)
```

Comment:
is this test redundant with the for condition oc_id != OC_UNKNOWN, or is it also hit in practice when iterating through daos_obj_classes[]? I don't think it hurts to have the check, just wondering since looking at daos_obj_classes[] value in the code it seems this condition may not happen?

Reply:
I was wondering if this test could be removed altogether and leave only the test that oc->oc_id != OC_UNKNOWN to control entry into the loop body. The only single (final) entry in daos_obj_classes with oc_name == NULL is also the only entry with oc_id == OC_UNKNOWN.

Reply:
Ok, ok, will remove it in next commit...